src/main/java/com/indix/client/impl/IndixApiClientImpl.java

```diff
@@ -245,6 +244,7 @@ public OffersSearchResult getProductsOffersPremium(Query query) throws IndixApiE
                     });
             return searchIndixApiResponse.getResult();
         } catch (IndixApiException iae) {
+            logger.error("getProductsOffersPremium failed: " + iae.getMessage());
             throw iae;
         } catch (Exception e) {
             logger.error("getProductsOffersPremium failed: " + e.getMessage());
```

Comment:
If any exception occurs for bad request/unauthorised request/too many requests/payment required then, we should throw the corresponding type of exception with the message, such that the user may handle them. But any other type of exception cannot be explicitly resolved by the user(connection error etc), and the hence can be clubbed as internal server error.